codegen/golang.go

```diff
@@ -33,13 +39,13 @@ func valueToSchemaType(v reflect.Type) (schema.ValueType, error) {
 	case reflect.Map:
 		return schema.TypeJSON, nil
 	case reflect.Struct:
-		t := v.PkgPath() + "." + v.Name()
-		if t == "time.Time" {
+		timeValue := time.Time{}
+		if v == reflect.TypeOf(timeValue) {
```

Comment:
This seems like a safer way to check for types instead of string comparison

Reply:
Im not sure this will work as `TypeOf` will return Struct

Reply:
cc @hermanschaaf I think battled with that recently

Reply:
There's a test case for this and looks like it passes 👍  I spent a few minutes a while back and couldn't figure out how to make it work, but this looks good!

Reply:
Ah, let me check it doesn't return `true` for all structs

Reply:
Ah yeah, that could be 😅

Reply:
>Im not sure this will work as TypeOf will return Struct

`reflect.TypeOf(timeValue)` returns an instance of `reflect.Type` so probably has all the information about the type to do the comparison

Reply:
But this change is unrelated to this PR, so I'll open another one for it

Reply:
Good idea, let's discuss in another PR. But I think it must work, since the test case includes a struct that gets converted correctly, as well as the `*time.Time` and `time.Time` cases.